client/db/bolt/db.go

```diff
@@ -437,12 +438,33 @@ func (db *BoltDB) UpdateOrder(m *dexdb.MetaOrder) error {
 	if len(md.Proof.DEXSig) == 0 {
 		return fmt.Errorf("cannot save order without DEX signature")
 	}
-	return db.ordersUpdate(func(master *bbolt.Bucket) error {
+	return db.ordersUpdate(func(ob, aob *bbolt.Bucket) error {
 		oid := ord.ID()
-		oBkt, err := master.CreateBucketIfNotExists(oid[:])
-		if err != nil {
-			return fmt.Errorf("order bucket error: %w", err)
+
+		// Create or move an order bucket based on order status. Orders
+		// that have status less than or equal to booked go in the
+		// ordersBucket. Finished orders go in the archivedOrdersBucket.
+		var oBkt *bbolt.Bucket
+		var err error
+		executed := md.Status > order.OrderStatusBooked
+		if executed {
+			if ob.Bucket(oid[:]) != nil {
+				ob.DeleteBucket(oid[:])
+			}
+			oBkt, err = aob.CreateBucketIfNotExists(oid[:])
+			if err != nil {
+				return fmt.Errorf("order bucket error: %w", err)
+			}
+		} else {
+			if aob.Bucket(oid[:]) != nil {
```

Comment:
This is expensive, and perhaps unnecessary.  If updating an active order requires an `aob.Bucket` call, we're really loosing a lot of the optimization we wanted from segregating archived orders.  Thoughts on failure modes and if we can live without this?

Reply:
I think the optimization was targeting reads and not writes. Even with it shouldn't affect getting active orders on startup. I can't imagine a situation where an order should go from archive to active, maybe a status unknown order?, but if one does this will create a duplicate order, which is different behavior than before. Since it probably won't happen, removing the check. Lets just not reverse order statuses...

Reply:
> I think the optimization was targeting reads

Right I'm was referring to `aob.Bucket(oid[:]) != nil` being expensive as it has to search through the archived orders bucket.